src/Command/Helper/TreeHelper.php

```diff
@@ -0,0 +1,117 @@
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP Project
+ * @since         5.3.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Command\Helper;
+
+use Cake\Console\Helper;
+use Closure;
+use UnitEnum;
+
+/**
+ * Tree command helper.
+ *
+ * Formats nested arrays into a tree with dashes and pipes.
+ */
+class TreeHelper extends Helper
+{
+    /**
+     * @inheritDoc
+     */
+    protected array $_defaultConfig = [
+        'initialIndent' => 0,
+    ];
+
+    /**
+     * Outputs an array in tree form.
+     *
+     * @param array $args Tree array
+     * @return void
+     */
+    public function output(array $args): void
+    {
+        $initialIndent = str_repeat(' ', $this->_config['initialIndent']);
+        foreach ($args as $key => $value) {
+            $this->outputElement($key, $value, $initialIndent, '', '');
+        }
+    }
+
+    /**
+     * Output an array in a tree.
+     *
+     * @param array $array
+     * @param string $prefix
+     * @return void
+     */
+    protected function outputArray(array $array, string $prefix): void
+    {
+        $i = 1;
+        $numValues = count($array);
+        foreach ($array as $key => $value) {
+            $isLast = $i++ === $numValues;
+            $marker = $isLast ? '└── ' : '├── ';
+            $indent = $isLast ? '    ' : '│   ';
+            $this->outputElement($key, $value, $prefix, $marker, $indent);
+        }
+    }
+
+    /**
+     * Output an array element.
+     *
+     * @param string|int $key
+     * @param mixed $value
+     * @param string $prefix
+     * @param string $marker
+     * @param string $indent
+     * @return void
+     */
+    protected function outputElement(
+        string|int $key,
+        mixed $value,
+        string $prefix,
+        string $marker,
+        string $indent,
+    ): void {
+        if (is_array($value)) {
+            $this->_io->out($prefix . $marker . $key);
+            $this->outputArray($value, $prefix . $indent);
+        } elseif (is_string($key)) {
+            $this->_io->out($prefix . $marker . $key);
+            $this->outputValue($value, $prefix . $indent . '└── ');
+        } else {
+            $this->outputValue($value, $prefix . $marker);
+        }
+    }
+
+    /**
+     * Output a value in a tree.
+     *
+     * @param mixed $value
+     * @param string $prefix
+     * @return void
+     */
+    protected function outputValue(mixed $value, string $prefix): void
+    {
+        if ($value instanceof Closure) {
+            $this->_io->out($prefix . $value());
+        } elseif ($value instanceof UnitEnum) {
```

Comment:
This is missing the similar checks we have in other places to have a more convenient output for labels if possible IMO.
See my linked PR

So `5` currently would be displayed as `Active`.

Reply:
if ($content instanceof EnumLabelInterface) {
					$content = $content->label();
				} elseif ($content instanceof BackedEnum) {
					$content = $content->value;
				} elseif ($content instanceof UnitEnum) {
					$content = $content->name;
				}

Reply:
I understand the idea but what is this tree we're trying to display?

Reply:
If you dont want to add it now thats fine, I can make follow up PRs.
The whole point of this is to not have to extend and customize in the first place :) A helper that helps instead of blocks people from displaying values I think would be the idea.

Reply:
For this specific example imagine one wants to display the users persmissions/roles.
It would be more cumbersome to have to correctly format than just passing the enums and know its gonna be formatted according to Cake principles.